core/components/commerce_timeslots/model/commerce_timeslots/commerce_timeslots.class.php

```diff
@@ -0,0 +1,150 @@
+ *
+ * It is built to be used with the MODX Revolution CMS.
+ *
+ * @category commerce
+ * @package commerce_timeslots
+ * @author Mark Hamstra <mark@modmore.com>
+ * @license See core/components/commerce/docs/license.txt
+ * @link https://www.modmore.com/commerce/
+ */
+class Commerce_Timeslots {
+
+    public $modx = null;
+    public $commerce = null;
+    public $adapter = null;
+    public $namespace = 'commerce_timeslots';
+    public $cache = null;
+    public $options = [];
+
+    public function __construct(modX &$modx, array $options = []) {
+        $this->modx =& $modx;
+        $this->namespace = $this->getOption('namespace', $options, 'commerce_timeslots');
+
+        $corePath = $this->getOption('core_path', $options, $this->modx->getOption('core_path', null, MODX_CORE_PATH) . 'components/commerce_timeslots/');
+        $assetsPath = $this->getOption('assets_path', $options, $this->modx->getOption('assets_path', null, MODX_ASSETS_PATH) . 'components/commerce_timeslots/');
+        $assetsUrl = $this->getOption('assets_url', $options, $this->modx->getOption('assets_url', null, MODX_ASSETS_URL) . 'components/commerce_timeslots/');
+
+        $this->options = array_merge([
+            'namespace' => $this->namespace,
+            'corePath' => $corePath,
+            'modelPath' => $corePath . 'model/',
+            'chunksPath' => $corePath . 'elements/chunks/',
+            'snippetsPath' => $corePath . 'elements/snippets/',
+            'templatesPath' => $corePath . 'templates/',
+            'assetsPath' => $assetsPath,
+            'assetsUrl' => $assetsUrl,
+            'jsUrl' => $assetsUrl . 'js/',
+            'cssUrl' => $assetsUrl . 'css/',
+            'connectorUrl' => $assetsUrl . 'connector.php'
+        ], $options);
+
+        // Load the Commerce service
+        $commercePath = $modx->getOption('commerce.core_path', null, MODX_CORE_PATH . 'components/commerce/');
+        $this->commerce = $this->modx->getService('commerce','Commerce',$commercePath . 'model/commerce/',$options);
+        if (!($this->commerce instanceof \Commerce)) $this->modx->log(modX::LOG_LEVEL_ERROR,'Couldn\'t load Commerce service!');
+
+        $this->adapter = $this->commerce->adapter;
+        $this->adapter->loadLexicon('commerce:default');
+
+        $this->adapter->loadPackage('commerce_timeslots', $this->getOption('modelPath'));
+        $this->adapter->loadLexicon('commerce_timeslots:default');
+    }
+
+    /**
+     * @param array $params
+     * @return string
+     */
+    public function getTimeslots(array $params): string
+    {
+        // Check if a custom template was provided - grid template is used by default
+        $tpl = $params['tpl'];
+        if(!$tpl) {
+            $tpl = 'timeslots/frontend/snippet_grid.twig';
+        }
+
+        // Check if a shipping method id was specified ( 0 returns all )
+        $shippingMethodId = $params['shippingMethod'] ?? 0;
```

Comment:
As the getShippingMethods typehints for `int`, I think this may also need to be cast to int to avoid errors as snippet properties are typically strings.